### PR TITLE
[MRG+1] DOC Replaced documentation with numpydoc for semilogx

### PR DESCRIPTION
**Pull request by DoriekeMG**
Hello matplotlib!

We have updated the documentation for the semilogx function to numpydoc.

Pavel Soriano & Dorieke Grijseels

@NelleV

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -1544,39 +1544,43 @@ def loglog(self, *args, **kwargs):
...
+	    Non-positive values in *x* can be masked as
+            invalid, or clipped to a very small positive number.
 
-        The remaining valid kwargs are
+	Returns
+	-------
+	`~matplotlib.pyplot.plot`
+	    Log-scaled plot on the *x* axis.
+
+	Other Parameters
```
Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: I think there is a slight indentation problem here. The bottom "---" don't start at the correct place.

Comment (NelleV) [inline lib/matplotlib/axes/_axes.py]: It's actually fine. my mistake

**Comment by NelleV** [comment]
Hello!

Your changes unfortunately contain tabs. The tests don't pass, and it also messes the sphinx rendering.
Do you mind editing the file again to remove the tabs?
If you come and see me before the end of the summer school, I can help you with that.

Cheers,
N

**Comment by QuLogic** [comment]
It appears you have not set the config for your git client. All your commits are attributed to "Your Name <you@example.com>", which is probably not what you want.

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -1544,39 +1544,43 @@ def loglog(self, *args, **kwargs):
...
+        Returns
+        -------
+        `~matplotlib.pyplot.plot`
+            Log-scaled plot on the *x* axis.
+
+        Other Parameters
+        ----------------
         :class:`~matplotlib.lines.Line2D` properties:
 
         %(Line2D)s
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: `Line2D` will need to be updated in order for this interpolation to work correctly.

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -1544,39 +1544,43 @@ def loglog(self, *args, **kwargs):
...
-
-          semilogx(*args, **kwargs)
-
-        :func:`semilogx` supports all the keyword arguments of
-        :func:`~matplotlib.pyplot.plot` and
-        :meth:`matplotlib.axes.Axes.set_xscale`.
-
-        Notable keyword arguments:
-
-          *basex*: scalar > 1
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: This condition should be noted in the new text.

**Comment by NelleV** [comment]
Wohoo! The tests pass!

**Comment by NelleV** [comment]
Here is how it looks (the top part, as my screen is too small to display the whole documentation):

semilogx

Thanks a lot! It's much better than the old version.

**Comment by QuLogic** [comment]
Note that the comments I've left here have already been handled, but GitHub doesn't have a "Mark Resolved" button for comments.

**Comment by QuLogic** [comment]
Backported to v2.x via f4152cefaff01621b025eaa6267e8122ea0e7c07.

**Pull request merged**